src/utils.ts

```diff
@@ -27,13 +27,25 @@ export const extendArray = <T>(arr1: T[], arr2: T[]): T[] => {
   return arr1;
 };
 
+const earliest = (a: number, b: number) => {
+  if (a === -1 && b === -1) return 0;
```

Comment:
comment: should we return `-1` in the case that there are no spaces or newlines? Since `0` is a valid index for `a` or `b`.

Reply:
`0` was the old default here so I think that should remain unchanged, otherwise the substring call we operate backwards